### PR TITLE
Created a new Autosuggest control that better navigates like the user expects

### PR DESCRIPTION
**Pull request by ztolley**
Removed a 3rd party autosuggest that just didn't act as the user expected the result is a much smaller and simpler component that we have control over and doesn't require yet more 3rd part elements.

Fixed node restart by excluding some folders
Reduced size of js files by 500k picking which parts of lodash to import.

### CODE REVIEW

**Comment by web-bert** [comment]
I think these designs make it clearer that you need to select one, and that the selections are not from the browser:

[Extend outline](https://designpatterns.hackpad.com/Type-ahead-l3OHvWUmDml#:h=Example)
[Type into new element](https://designpatterns.hackpad.com/International-country-selector-14jGm6DC3YE#:h=Possible-solution-for-interfac)

**Comment by web-bert** [comment]
Much better with the validation though

File: app/javascripts/components/autosuggest.component.js [hunk 1]
```diff
@@ -1,73 +1,155 @@
...
     }
 
     if (props.value) {
-      state.selected = props.value;
+      state.value = props.value;
     }
 
     this.state = state;
-    this.fetchSuggestionsFromServer = debounce(this.fetchSuggestionsFromServer, 300);
+    this.fetchSuggestions = debounce(this.fetchSuggestions, 200);
```
Comment (web-bert) [inline app/javascripts/components/autosuggest.component.js]: Debounce only needed when fetching from a server

File: app/javascripts/components/autosuggest.component.js [hunk 2]
```diff
@@ -78,7 +160,9 @@ export class AutosuggestComponent extends Component {
     if (this.props.fetchSuggestions) {
```
Comment (web-bert) [inline app/javascripts/components/autosuggest.component.js]: Put this logic in the constructor and only debounce for server calls

File: app/javascripts/components/didyoumeancompany.component.js [hunk 3]
```diff
@@ -14,62 +10,37 @@ export class DidYouMeanCompanyComponent extends Component {
...
-  renderSuggestion = (suggestion) => {
-    return (
-      <div className="autosuggest__suggestion">
-        <Highlight search={this.state.selected.name}>
-          {suggestion.name}
-        </Highlight>
-      </div>
-    );
+  fetchSuggestions = (value) => {
+    return new Promise((fullfill) => {
```
Comment (web-bert) [inline app/javascripts/components/didyoumeancompany.component.js]: I think the convention is to call this "resolve", and it should only have one L "fulfill" or even "fulfil" ;-)

[Spec](https://tc39.github.io/ecma262/#sec-promise-objects)
[Google dev](https://developers.google.com/web/fundamentals/getting-started/primers/promises)

File: app/javascripts/forms/interactionform.js [hunk 4]
```diff
@@ -102,15 +102,18 @@ export class InteractionForm extends BaseForm {
...
-      .then(response => {
-        return response.data.map(({id, first_name, last_name}) => {
-          return {
-            id,
-            name: `${first_name} ${last_name}`
-          };
+    return new Promise((fullfill) => {
+      axios.get(`/api/contactlookup?company=${this.state.formData.company.id}&contact=${term}`)
+        .then(response => {
+          const result = response.data.map(({id, first_name, last_name}) => {
```
Comment (web-bert) [inline app/javascripts/forms/interactionform.js]: Do this mapping in Node?

File: app/javascripts/components/autosuggest.component.js (hunk 1 above)
Comment (ztolley) [inline app/javascripts/components/autosuggest.component.js]: I did originally have this but the mix of bounced and debounced caused problems. Will give it another go but did struggle with that for a while.

File: app/javascripts/components/didyoumeancompany.component.js (hunk 3 above)
Comment (ztolley) [inline app/javascripts/components/didyoumeancompany.component.js]: Agreed. we seem to have a mix and that doesn't help.

File: app/javascripts/forms/interactionform.js (hunk 4 above)
Comment (ztolley) [inline app/javascripts/forms/interactionform.js]: Agreed.

**Pull request merged**